sdk/state/integrationtests/state_test.go

```diff
@@ -42,37 +42,69 @@ func TestOpenUpdatesUncoordinatedClose(t *testing.T) {
 
 	// Open
 	t.Log("Open...")
+	// I signs txClose
 	open, err := initiatorChannel.ProposeOpen(state.OpenParams{Asset: asset, AssetLimit: assetLimit})
 	require.NoError(t, err)
-	for {
+	assert.ElementsMatch(t,
+		[]int{1, 0, 0},
+		[]int{len(open.CloseSignatures), len(open.DeclarationSignatures), len(open.FormationSignatures)},
+	)
+	{
 		var authorizedR bool
+		// R signs txClose and txDecl
 		open, authorizedR, err = responderChannel.ConfirmOpen(open)
-		if err != nil {
-			t.Fatal(err)
-		}
+		require.NoError(t, err)
+		require.False(t, authorizedR)
+		assert.ElementsMatch(t,
+			[]int{2, 1, 0},
+			[]int{len(open.CloseSignatures), len(open.DeclarationSignatures), len(open.FormationSignatures)},
+		)
+
 		var authorizedI bool
+		// I signs txDecl and F
 		open, authorizedI, err = initiatorChannel.ConfirmOpen(open)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if authorizedI && authorizedR {
-			break
-		}
+		require.NoError(t, err)
+		require.False(t, authorizedI)
+		assert.ElementsMatch(t,
+			[]int{2, 2, 1},
+			[]int{len(open.CloseSignatures), len(open.DeclarationSignatures), len(open.FormationSignatures)},
+		)
+
+		// R signs F, R is done
+		open, authorizedR, err = responderChannel.ConfirmOpen(open)
+		require.NoError(t, err)
+		require.True(t, authorizedR)
+		assert.ElementsMatch(t,
+			[]int{2, 2, 2},
+			[]int{len(open.CloseSignatures), len(open.DeclarationSignatures), len(open.FormationSignatures)},
+		)
+
+		// I receives the last signatures for F, I is done
+		open, authorizedI, err = initiatorChannel.ConfirmOpen(open)
+		require.NoError(t, err)
+		require.True(t, authorizedI)
+		assert.ElementsMatch(t,
+			[]int{2, 2, 2},
+			[]int{len(open.CloseSignatures), len(open.DeclarationSignatures), len(open.FormationSignatures)},
+		)
 	}
 
 	{
-		ci, di, fi, err := initiatorChannel.OpenTxs(state.OpenParams{Asset: asset, AssetLimit: assetLimit})
+		ci, di, fi, err := initiatorChannel.OpenTxs(state.OpenParams{
+			Asset:      initiatorChannel.OpenAgreement().Details.Asset,
+			AssetLimit: initiatorChannel.OpenAgreement().Details.AssetLimit,
+		})
```

Comment:
💭 I think we need to improve how outside code gets at the the txs, but we don't need to solve that here. I opened a separate issue about it: https://github.com/stellar/experimental-payment-channels/issues/127.